### PR TITLE
Update lockfiles and fix for new esmpy

### PR DESCRIPTION
**Pull request by stephenworsley**
Replaces #335 and adds a fix.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -370,6 +370,7 @@ def benchmarks(
     COMPARE_FACTOR = 2.0
 
     session.install("asv", "nox", "pyyaml")
+    session.run("conda", "install", "--yes", "conda<24.3")
```
Comment (bjlittle) [inline noxfile.py:373]: @stephenworsley Consider making an issue to remind ourselves to unpin this whenever `asv` updates to not use the `--force` flag 👍

**Comment by schlunma** [comment]
Hey @stephenworsley, I just did some updates to my environment and started to see issues like

```
ValueError: ESMC_MeshAddElement() failed with rc = 509.    Please check the log files (named "*ESMF_LogFile")
```

when using iris-esmf-regrid. This only appears with `esmpy=8.6.1`; with `esmpy=8.4.2` I don't get this error. Does this PR here fix that? Any chance there will be a new release of iris-esmf-regrid soon? Thanks!!

**Comment by stephenworsley** [comment]
@schlunma Yes, I believe this should fix this error, I'm thinking of doing a release at some point at the end this month.

**Comment by valeriupredoi** [comment]
> Hey @stephenworsley, I just did some updates to my environment and started to see issues like
> 
> ```
> ValueError: ESMC_MeshAddElement() failed with rc = 509.    Please check the log files (named "*ESMF_LogFile")
> ```
> 
> when using iris-esmf-regrid. This only appears with `esmpy=8.6.1`; with `esmpy=8.4.2` I don't get this error. Does this PR here fix that? Any chance there will be a new release of iris-esmf-regrid soon? Thanks!!

saw exactly the same behaviour elsewhere (ANTS) two days ago :+1:

**Comment by bouweandela** [comment]
I see no changes to the iris-esmf-regrid code in this pull request, only to test infrastructure. How can this fix the issue described by @schlunma and why would a new release help if the released code is identical?

**Comment by stephenworsley** [comment]
@bouweandela this PR ended up getting split in two. The actual fix went into #353 in order for the benchmarks to be able to run before and after the lockfile update.

**Comment by bouweandela** [comment]
Thanks, that makes sense. Does the new code work with all versions of ESMPy? Or just 8.6 and up?

**Comment by schlunma** [comment]
> Thanks, that makes sense. Does the new code work with all versions of ESMPy? Or just 8.6 and up?

@stephenworsley sorry to bother you again, but would you know if if this new version is also compatible with ESMPy < 8.6.0?

**Comment by valeriupredoi** [comment]
> > Thanks, that makes sense. Does the new code work with all versions of ESMPy? Or just 8.6 and up?
> 
> @stephenworsley sorry to bother you again, but would you know if if this new version is also compatible with ESMPy < 8.6.0?

same q from me too, please, folks :beer:

**Pull request merged**